docs/patterns.rst

```diff
@@ -15,8 +15,71 @@ Slots are for storing information that's relevant over multiple turns. For examp
 our restaurant example, we would want to keep track of things like the cuisine and number of 
 people for the duration of the conversation. 
 
-Collecting Information to Complete a Request
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Slot Filling
+^^^^^^^^^^^^
+
+Probably the most common pattern of all is to collect a few pieces of information
+from a user in order to do something (book a restaurant, call an API, search a database, etc.).
+This is also called slot filling. 
+
+To make this easier, Rasa has a special action class called ``FormAction``.
+This lets you have a single action for this task, rather than separate actions for each question,
+e.g. ``utter_ask_cuisine``, ``utter_ask_numpeople``, etc. 
+
+You don't *have* to use ``FormAction`` s to do slot filling, but it can be easier. 
+
+A form action has a set of required fields, which you define for the class:
+
+.. code-block:: python
+
+   class RestaurantSearch(FormAction):
+   
+       REQUIRED_FIELDS = [
+	  EntityFormField("number", "num_people"),
+          EntityFormField("cuisine", "cuisine")
+       ]
+
+       def name(self):
+           return 'action_search_restaurants'
+
+       def submit(self, dispatcher, tracker, domain):
+           results = RestaurantAPI.search(tracker.get_slot("cuisine"),
+                                         tracker.get_slot("num_people"))
+           return [SlotSet("results", results)]
+
+
+The way this works is that every time you call this action, it will pick one of the 
+``REQUIRED_FIELDS`` that's still missing and ask the user for it. You can also ask a yes/no
+question with a ``BooleanFormField``.
+
+The story will look something like this:
+
+.. code-block:: md
+
+   * request_restaurant
+        - action_restaurant_form
+        - slot{"requested_slot": "people"}
+   * inform{"number": 3}
+        - action_restaurant_form
+        - slot{"people": 3}
+        - slot{"requested_slot": "time"}
+   * inform{"time": "8pm"}
+      - action_restaurant_form
+
+
+Some important things to consider:
+
+- Your domain needs to have a slot called ``requested_slot``. This can be unfeaturized, but if you want
+  to support contextual questions like *"why do you need to know that information?"* you should make this
+  a categorical slot. 
+- You need to define utterances for asking for each slot in your domain, e.g. ``utter_ask_{slot_name}``.
+- We strongly recommend that you create these stories using interactive learning, because if you
+  type these by hand you will probably forget to add all of the slots.
+- Any slots that are already set won't be asked for. E.g. if someone says "I'd like a Chinese restaurant for 8 people" the ``submit`` function should get called right away.
```

Comment:
would it make sense to briefly comment on the use of the `validate` method as well?